core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -391,8 +455,19 @@ public void requestConnection() {
         scheduleNextConnection();
         break;
       case TRANSIENT_FAILURE:
+        if (scheduleConnectionTask != null) {
```

Comment:
Why was this added? scheduleConnectionTask is guaranteed to be null for serializing retry, so this seems an unrelated bug or unnecessary.